packages/devtools_app/lib/src/screens/network/network_controller.dart

```diff
@@ -110,37 +91,28 @@ class NetworkController extends DisposableController
     List<SocketStatistic> sockets,
     List<HttpProfileRequest>? httpRequests,
     int timelineMicrosOffset, {
-    required List<NetworkRequest> currentValues,
+    required CurrentNetworkRequests currentRequests,
     required List<DartIOHttpRequestData> invalidRequests,
-    required Map<String, DartIOHttpRequestData> outstandingRequestsMap,
   }) {
-    // [currentValues] contains all the current requests we have in the
-    // profiler, which will contain web socket requests if they exist. The new
-    // [sockets] may contain web sockets with the same ids as ones we already
-    // have, so we remove the current web sockets and replace them with updated
-    // data.
-    currentValues.removeWhere((value) => value is WebSocket);
-    for (final socket in sockets) {
-      final webSocket = WebSocket(socket, timelineMicrosOffset);
-      // If we have updated data for the selected web socket, update the value.
-      if (selectedRequest.value is WebSocket &&
-          (selectedRequest.value as WebSocket).id == webSocket.id) {
-        selectedRequest.value = webSocket;
-      }
-      currentValues.add(webSocket);
+    currentRequests.updateWebSocketRequests(sockets, timelineMicrosOffset);
+
+    // If we have updated data for the selected web socket, we need to update
+    // the value.
+    final currentSelectedRequestId = selectedRequest.value?.id;
+    if (currentSelectedRequestId != null) {
+      selectedRequest.value =
+          currentRequests.getRequest(currentSelectedRequestId);
     }
```

Comment:
should this be done after we process the http profile requests? seems odd to be updating the selected value only after updating web socket requests

Reply:
we could, but since we only merge when http profile requests get updated, then the instance remains the same.
So it will still be pointing to the correct instance